phonenumberfmt/format_phone_number_list.py

```diff
@@ -0,0 +1,161 @@
+import collections
+import re
+
+from typing import List, Union
+from urllib.parse import unquote
+
+import phonenumbers
+
+from phonenumbers import PhoneNumberFormat
+from phonenumbers.phonenumberutil import NumberParseException
+
+from .country import dialing_prefix_for_region
+
+
+def _is_valid(phone_number: str, dialing_prefix: str) -> bool:
+    """Gets whether the specified phone number is a valid one."""
+    try:
+        parsed_number = phonenumbers.parse(phone_number, region=dialing_prefix)
+        if phonenumbers.is_possible_number(parsed_number) is True:
+            return True
+
+        if phonenumbers.is_valid_number(parsed_number) is True:
+            return True
+    except NumberParseException:
+        pass
+
+    return False
+
+
+def format_phone_number_list(
```

Comment:
Is it possible to split the functionality in simpler and focused functions? It seams this one is doing 3 things: it splits the input, validates the parts and then formats. I think that refactoring this in 2-3 functions would give more freedom to all the users of the library. Here is an example of how I would like to use this library as an external user:
```python
from phonenumberfmt import format_phone_number, is_valid_phone_number, split_phone_numbers

numbers = "+97144220125:40,+971503292710:40"

formatted_phone_numbers = [
    format_phone_number(number)
    for number in split_phone_numbers(numbers)
    if is_valid_phone_number(number)
]
```